internal/services/storagecache/managed_lustre_file_system_auto_export_job_resource.go

```diff
@@ -0,0 +1,235 @@
+	"github.com/hashicorp/go-azure-sdk/resource-manager/storagecache/2024-07-01/autoexportjob"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/storagecache/2024-07-01/autoexportjobs"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/storagecache/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type ManagedLustreFileSystemAutoExportJobModel struct {
+	Name               string            `tfschema:"name"`
+	ResourceGroupName  string            `tfschema:"resource_group_name"`
+	AmlFileSystemName  string            `tfschema:"aml_file_system_name"`
+	Location           string            `tfschema:"location"`
+	AutoExportPrefixes []string          `tfschema:"auto_export_prefixes"`
+	AdminStatus        string            `tfschema:"admin_status"`
+	Tags               map[string]string `tfschema:"tags"`
+}
+
+type ManagedLustreFileSystemAutoExportJobResource struct{}
+
+var _ sdk.ResourceWithUpdate = ManagedLustreFileSystemAutoExportJobResource{}
+
+func (r ManagedLustreFileSystemAutoExportJobResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r ManagedLustreFileSystemAutoExportJobResource) ModelObject() interface{} {
+	return &ManagedLustreFileSystemAutoExportJobModel{}
+}
+
+func (r ManagedLustreFileSystemAutoExportJobResource) ResourceType() string {
+	return "azurerm_managed_lustre_file_system_auto_export_job"
+}
+
+func (r ManagedLustreFileSystemAutoExportJobResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[0-9a-zA-Z][-0-9a-zA-Z_]{0,78}[0-9a-zA-Z]$`), "name must be 3-80 characters long and can only contain alphanumeric characters, underscores, and hyphens, and must start and end with an alphanumeric character"),
+		},
+
+		"location": commonschema.Location(),
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"aml_file_system_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ManagedLustreFileSystemName,
+		},
+
+		"auto_export_prefixes": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+
+		"admin_status": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(autoexportjob.AutoExportJobAdminStatusEnable),
+				string(autoexportjob.AutoExportJobAdminStatusDisable),
+			}, false),
+			Default: string(autoexportjob.AutoExportJobAdminStatusEnable),
+		},
```

Comment:
since the only possible values for this are `Enable` and `Disable`, should we consider making this a bool called  `admin_status_enabled` instead?